### ISSUE TITLE
SNN (Self-Normalizing Networks)

### ISSUE THREAD

**Issue by mdraw** [ELEKTRONN/ELEKTRONN2#12]
https://arxiv.org/abs/1706.02515 proposes a new activation function SELU, which can replace explicit batch normalization and could improve overall performance and accuracy of our neural networks.
To do:
- [x] Implement SELU activation function
- [ ] Implement special "alpha dropout" for SNNs because regular dropout is broken with SELU (see p. 6 of paper). I think we can do this transparently by automatically switching to the alpha dropout algorithm in `Perceptron` constructors if they detect `activation_func='selu'`
- [ ] Weight initialisation method suggested at the bottom of p.3 of the paper (&rarr; `elektronn2.neuromancer.variables.initweights()`)

Links:
- [General discussion](https://www.reddit.com/r/MachineLearning/comments/6g5tg1/r_selfnormalizing_neural_networks_improved_elu/)
- [Practical discussion of issues with an SNN U-Net variant](http://forums.fast.ai/t/self-normalizing-neural-networks/3558)
- https://github.com/bioinf-jku/SNNs
- https://github.com/fchollet/keras/pull/6969
